src/apps/answers/service.py

```diff
@@ -219,6 +219,34 @@ async def _validate_applet_for_user_response(self, applet_id: uuid.UUID) -> None
         if not roles:
             raise UserDoesNotHavePermissionError()
 
+    async def _validate_temp_take_now_relation_between_subjects(
+        self, source_subject_id: uuid.UUID, target_subject_id: uuid.UUID
+    ) -> None:
+        relation = await SubjectsCrud(self.session).get_relation(source_subject_id, target_subject_id)
+
+        if is_take_now_relation(relation) and not is_valid_take_now_relation(relation):
+            raise ValidationError("Invalid temp take now relation between subjects")
```

Comment:
This is correct validation, but it's being done between the wrong subjects. It should be done between the:
- respondent and source
- respondent and target